### PR TITLE
Chk 3279 change stream

### PR DESCRIPTION
**Pull request by CianoDanilo**
#### Motivation and Context
This is the first implementation of a change data capture mechanism for the payment wallet domain.
This application will listen to the wallet log events collection for change events and then sends them to a queue.

#### How Has This Been Tested?
It has been tested with jUnit tests and monitored after release in DEV env 

#### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as
  expected)

#### Checklist:

- [ ] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.

### CODE REVIEW

File: build.gradle.kts
Comment (EmanueleBVtech) [inline build.gradle.kts:1]: why not use mockito v5.4.0 (latest)?

Comment (CianoDanilo) [reply build.gradle.kts:1]: Didn't have any trouble with the current version. If you solved the problems related to ByteBuddy and Java 21 compatibility, I can upgrade the version, resolving a merge conflict ahead of time, let me know

Comment (EmanueleBVtech) [reply build.gradle.kts:1]: the problem with bytebuddy has been solved by adding explicit reference to the dependency
implementation("net.bytebuddy:byte-buddy:1.15.3")

about mockito v.5.4.0, it was just a suggestion so we are using the latest library version, this is not a strict requirement

Comment (CianoDanilo) [reply build.gradle.kts:1]: Dependency updated in [cac9001](https://github.com/pagopa/pagopa-payment-wallet-cdc-service/pull/6/commits/cac9001a0a8987b38d00248a8a1e8a9de4b83b9c)

**Review by EmanueleBVtech — DISMISSED** [review]

File: gradle.lockfile [hunk 1]
```diff
@@ -1,107 +0,0 @@
-# This is a Gradle generated file for dependency locking.
```
Comment (infantesimone) [inline gradle.lockfile:1]: @CianoDanilo why delete entire file?

**Comment by infantesimone** [comment]
to reduce the number of changed lines in a PR, in general, if you encounter errors/bug that are not related to PR, it is better to resolve them in dedicated PR (I see changes to the pipeline configuration and helm values ​​in uat)

File: build.gradle.kts [hunk 2]
```diff
@@ -32,6 +33,8 @@ dependencyLocking { lockAllConfigurations() }
 
 dependencies {
   implementation("org.springframework.boot:spring-boot-starter-data-mongodb-reactive")
+  implementation("org.springframework.boot:spring-boot-starter-actuator")
```
Comment (infantesimone) [inline build.gradle.kts:36]: why introduced in this PR?

Comment (CianoDanilo) [reply build.gradle.kts:36]: The actuator was needed for the readiness and liveness probes, next time I'll handle this kind of work in a separated PR

File: helm/pay-wallet-values-dev.yaml [hunk 3]
```diff
@@ -76,8 +76,27 @@ microservice-chart:
...
+    MONGO_CONNECTION_TIMEOUT_MS: "2000"
+    MONGO_SOCKET_TIMEOUT_MS: "10000"
+    MONGO_SERVER_SELECTION_TIMEOUT_MS: "2000"
+    MONGO_WAITING_QUEUE_MS: "2000"
+    MONGO_HEARTBEAT_FREQUENCY_MS: "5000"
+    CDC_LOG_EVENTS_COLLECTION_NAME: "payment-wallets-log-events"
+    CDC_LOG_EVENTS_OPERATION_TYPE: "insert,update,replace"
+    CDC_LOG_EVENTS_PROJECT: "fullDocument"
+    CDC_SEND_RETRY_MAX_ATTEMPTS: 1
+    CDC_SEND_RETRY_INTERVAL_IN_MILLIS: 1000
```
[the remark this answers is not in the document]
Comment (CianoDanilo) [reply helm/pay-wallet-values-dev.yaml]: Renamed in [fbb18b0](https://github.com/pagopa/pagopa-payment-wallet-cdc-service/pull/6/commits/fbb18b0538277f2cfe5fa46e848f7e4a39b73e51)

File: gradle.lockfile (hunk 1 above)
Comment (CianoDanilo) [reply gradle.lockfile]: File added again in [fbb18b0](https://github.com/pagopa/pagopa-payment-wallet-cdc-service/pull/6/commits/fbb18b0538277f2cfe5fa46e848f7e4a39b73e51)

**Review by infantesimone — APPROVED** [review]
great! 🚀🚀🚀

**Pull request merged**